sympy/polys/polytools.py

```diff
@@ -4404,13 +4404,20 @@ def degree(f, *gens, **args):
 
     """
     options.allowed_flags(args, ['gen', 'polys'])
-
+    f = sympify(f).as_expr()
```

Comment:
If `f` is a poly, it might not have the gen for the variable whose degree we need (and poly_from_expr will retain the gens that the input has). For instance, suppose we call `degree(Poly(x**2+y**2, x), y)`. `poly_from_expr(Poly(x**2+y**2,x))[0]` gives back `Poly(x**2+y**2,x)` (same gens). If we tried to then call `Poly(x**2+y**2,x).degree(y)`, it would give an error because y isn't one of the gens.